src/compactor_executor.rs

```diff
@@ -123,7 +125,30 @@ impl TokioCompactionExecutorInner {
             .table_store
             .table_writer(SsTableId::Compacted(Ulid::new()));
         let mut current_size = 0usize;
-        while let Some(kv) = all_iter.next_entry().await? {
+        let now = self.options.clock.now();
+
+        while let Some(raw_kv) = all_iter.next_entry().await? {
+            // filter out any expired entries -- eventually we can consider
+            // abstracting this away into generic, pluggable compaction filters
+            // but for now we do it inline
+            let kv = match raw_kv.attributes.expire_ts {
+                None => raw_kv,
+                Some(expire_ts) => {
```

Comment:
You might be able to do `Some(expire_ts) if expire_ts <= now` and then have a `_ => raw_kv` otherwise.

Reply:
that's cool, didn't realize you could have conditions like that in `match`